### ISSUE TITLE
Question on terminal setup

### ISSUE THREAD

**Issue by bk42327** [MattisLind/DP2200#4]
Hi Matis,

I may have missed it but I found no other means to contact you, not sure if you are following the other forum.,

The current terminal setup is useful mostly, what I would suggest is a simple similar setup. What do you think to run the initial control session where you load tapes etc in a default terminal and spawn 2 more windows for console and debug.

That would allow a bit more flexibility but I also understand the complexity as this would require a serious code change...

Other than that, I see this progresses and you do a fabulous job, in case you want to start at some point with a little documentation I am herewith volunteering to help and add whatever is left from my memory.

Cheers and many thanks for your contribution.
Bernd

**Comment by MattisLind** [comment]
Hello Bernd!

I have been thinking a bit on how to organize windows. Mostly because I want a possibility to have a X-terminal window for the 2200 console where I can use the software loadable fonts that the later display card made possible.

When doing that it might be possible to add a separare debugging window. 

I will definitely take it into consideration, but it will not happen soon.

On the other hand I have been thinking of adding support for script files so that you can add a script file as an argument ot start it from the command line. Thereby setting everything up and then just run a scenario.

Regarding documentation; I haven't focused much on documenting the simulator. Just trying to add some simple command line help when adding a new command.

Feel free to make a pull request towards the repo or just send me files (in markdown format preferbly) and I can add it!

What is the forum where you are discussing the simulator?

BTW. Right now I am fighting to get it to load the DOS.D 2.6. I get so far that it tries to load stuff on the 9370 hard drive, but then something fails. Haven't figured out what the problem is. 

I am looking for any documentation on the 9370 disk drive. The documentation on it is rather scarce.

**Comment by bk42327** [comment]
Yes, DOS.D is the deal breaker here, we need Databus compiler and Datashare to run. Most business application were programs running on the terminals (the console was much more for batch and systems operations). That said,  a terminal window would be pretty cool.

Cheers,
Bernd

**Comment by MattisLind** [comment]
I have been thinking on creating an emulated 8 port serial board as well. Then each port is emulated using a TCP/IP port to which you could use telnet to connect to. Someone has to do a terminal emulator that emulates those terminals that were used (Datapoint 3300 / 3600).

Yes, it would be great to get DOS.D going. But right now I am kind of stuck since it claims the disk is out of space.

<img width="1433" alt="Screenshot 2024-02-06 at 09 42 20" src="https://github.com/MattisLind/DP2200/assets/4656060/d6c26f52-e88f-477d-8b34-1e965a34ec7d">

The last thing it does is to read a sector of the disk and comparing each byte with the value 150 octal. It does this twice and then prints the message. I wonder if it is something with the 9370 disk that I don't really know about. The only info on the 9370 is a single text page with register description. It is a bit scarce...

Cannot figure out what it expect to be there and why it compare with 150 octal. I will add some more tracing to disk writes and disk reads to see if there is bugs in there. 

Any help is highly appreciated!

**Comment by bk42327** [comment]
Here are 2 docs I had which reference the disk drives. You may have them already.

When I worked with Datapoint the inards of the system were well protected, mostly hardware technicians were educated.
I.e. the SNAP assembler compiler was closely guarded.


> On 6. Feb 2024, at 08:29, Bernd Kunze ***@***.***> wrote:￼￼
> Yes, DOS.D is the deal breaker here, we need Databus compiler and Datashare to run. Most business application were programs running on the terminals (the console was much more for batch and systems operations). That said,  a terminal window would be pretty cool.
> 
> Cheers,
> Bernd
>

**Comment by MattisLind** [comment]
Hello Bernd!

Your document didn't make it through github. You may contact me on mattis (commercial-at replace with @) mattisborgen (dot) se

I have access to all the stuff at bitsavers. If there is something else then I don't have it. For the 9350 there is a quite detailed document: [9350 prod spec](http://www.bitsavers.org/pdf/datapoint/prodSpec/2200-350_diabloDisk_Mar72.pdf). But unfortunately there is no such document for the 9370.
